File: myrealm-classes/myrealm-model.ts
import * as Realm from "realm";

export type StaticColor = {};

export const StaticColorSchema = {
  name: "StaticColor",
  properties: {},
};

export type Task = {
  _id: number;
  name: string;
  status?: string;
};

export const TaskSchema = {
  name: "Task",
  primaryKey: "_id",
  properties: {
    _id: "int",
    name: "string",
    status: "string?",
  },
};

export type Test = {
  _id: Realm.ObjectId;
  color?: Task;
};

export const TestSchema = {
  name: "Test",
  primaryKey: "_id",
  properties: {
    _id: "objectId",
    color: "Task",
  },
};

export const Schema = [StaticColorSchema, TaskSchema, TestSchema];

File: myrealm-classes/myrealm-model.swift
import Foundation
import RealmSwift

class Static Color: Object {
}

class Task: Object {
    @objc dynamic var _id: Int = 0
    @objc dynamic var name: String = ""
    @objc dynamic var status: String? = nil

    override static func primaryKey() -> String? {
        return "_id"
    }
}

class Test: Object {
    @objc dynamic var _id: ObjectId = ObjectId()
    @objc dynamic var color: Task?

    override static func primaryKey() -> String? {
        return "_id"
    }
}

